# Question

## Which category is the least successful one?

# 1 Load your data

In [1]:
# Import Python packages needed for your Data Exploration
import pandas as pd

In [2]:
# Floats (decimal numbers) should be displayed rounded with 2 decimal places
pd.options.display.float_format = "{:,.2f}".format

In [3]:
# Import your data to a DatFrame format
dataframe = pd.read_excel("orders_clean.xlsx")

### Your data is  stored in a variable, called 'dataframe'.  
### You can use different attributes and methods on your DataFrame object.



# 2 Understand your data

## How does your dataframe look like?

In [4]:
# with '.head()' the first x rows of your dataframe get displayed
dataframe.head(3)

Order_ID Order_Date  Ship_Date     Ship_Mode Customer_ID  \
0  CA-2020-152156 2020-11-08 2020-11-11  Second Class    CG-12520   
1  CA-2020-152156 2020-11-08 2020-11-11  Second Class    CG-12520   
2  CA-2020-138688 2020-06-12 2020-06-16  Second Class    DV-13045   

  Country/Region         City       State Region       Product_ID  \
0  United States    Henderson    Kentucky  South  FUR-BO-10001798   
1  United States    Henderson    Kentucky  South  FUR-CH-10000454   
2  United States  Los Angeles  California   West  OFF-LA-10000240   

          Category Sub_Category  \
0        Furniture    Bookcases   
1        Furniture       Chairs   
2  Office Supplies       Labels   

                                        Product_Name  Sales  Quantity  \
0                  Bush Somerset Collection Bookcase 261.96         2   
1  Hon Deluxe Fabric Upholstered Stacking Chairs,... 731.94         3   
2  Self-Adhesive Address Labels for Typewriters b...  14.62         2   

   Discount  Profit  
0      0.00   41.91  
1      0.00  219.58  
2      0.00    6.87

## How many rows and columns does your dataframe consist of?

In [5]:
# use the '.shape' attribute to get information about number of rows and columns
dataframe.shape

(9994, 17)

### Our dataframe consists of 17 columns and 9994 rows.

## What are the summary statistics of Sales and Profit?

In [6]:
# the function '.describe' gives you a good overview of the descriptive stats of your numeric columns
dataframe[['Sales', 'Profit']].describe()

Sales    Profit
count  9,994.00  9,994.00
mean     229.86     28.66
std      623.25    234.26
min        0.44 -6,599.98
25%       17.28      1.73
50%       54.49      8.67
75%      209.94     29.36
max   22,638.48  8,399.98

## Which categories do exist?

In [7]:
# use '.unique' on the Category column to get an overview of your unique categories
dataframe['Category'].unique()

array(['Furniture', 'Office Supplies', 'Technology'], dtype=object)

## Your products belong to one of three different categories: Furniture, Office Supplies or Technology.

# 3 Answer questions

## Which category is the least successful one?

### a) ... with success is defined as sum of sales

In [8]:
dataframe.groupby(['Category'])['Sales'].sum().sort_values(ascending=False)

Category
Technology        836,154.03
Furniture         741,999.80
Office Supplies   719,047.03
Name: Sales, dtype: float64

### When looking at sum of sales per category, Office Supplies performs worst.

### b) ... with success defined as sum of profit

In [9]:
dataframe.groupby(['Category'])['Profit'].sum().sort_values(ascending=False)

Category
Technology        145,454.95
Office Supplies   122,490.80
Furniture          18,451.27
Name: Profit, dtype: float64

### When looking at sum of profit per category, Furniture performs worst.

## Okay, let us remove the category furniture - or not???

# 4 Question your findings

## Furniture performs the worst - should we remove the category and focus only on the two remaining ones?

## No!!! Within furniture, there might be sub-categories which show a good performance.


# 5 Iterate

#### Go back to Step 3 and look at profit per sub-category
#### Repeat Step 4 and question your Insights: Should we remove underperforming sub-categories?
###  No!!! Within these sub-categories, there might be products which show a good performance.
#### Go back to Step 3 and look at profit per product
#### Repeat Step 4 and question your insights: Should we remove underperforming products?
.....